tfx/experimental/pipeline_testing/stub_launcher_test.py

```diff
@@ -0,0 +1,175 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for tfx.components.trainer.executor."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import mock
+import os
+from typing import Any, Dict, List, Text
+
+import absl
+import tensorflow as tf
+from ml_metadata.proto import metadata_store_pb2
+
+from tfx import types
+from tfx.experimental.pipeline_testing import stub_component_launcher
+from tfx.experimental.pipeline_testing import base_stub_executor
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.orchestration import publisher
+from tfx.orchestration.launcher import test_utils
+from tfx.types import channel_utils
+from tfx.utils import io_utils
+
+class CustomStubExecutor(base_stub_executor.BaseStubExecutor):
+  """Example of custom stub executor."""
+  def Do(self, input_dict: Dict[Text, List[types.Artifact]],
```

Comment:
It is unclear what's the purpose of a custom stub executor to me: parent class's `Do` function is not even called.

Reply:
Could you elaborate more about the expected behavior of this class? I don't think that we need to call super().Do here.